x-pack/plugins/reporting/server/lib/tasks/execute_report.ts

```diff
@@ -251,7 +255,17 @@ export class ExecuteReportTask implements ReportingTask {
     this.logger.debug(`Saving ${report.jobtype} to ${docId}.`);
 
     const completedTime = moment().toISOString();
-    const docOutput = this._formatOutput(output);
+    const { content, ...docOutput } = this._formatOutput(output);
+    const stream = await this.getContentStream({
+      id: report._id,
+      index: report._index!,
+      if_primary_term: report._primary_term,
+      if_seq_no: report._seq_no,
+    });
```

Comment:
I had in mind that the abstracted file storage mechanism would be used inside of the task runner functions, (aka `execute_job` functions) so they can take control of streaming their output content to storage as it becomes available.

Internally, the file storage mechanism could chunk up the data into multiple documents as it is available, which lends itself towards solving https://github.com/elastic/kibana/issues/18322

It might be good to create a `stream` variable in `_performJob`, and pass it to the task runner functions on line 247. We can have the task runners return the stream back again so we have the latest `_seq_no` and `_primary_term`. Then we could remove the parts from `execute_report.ts` that handle the entire output content.

Reply:
i agree we should be writing inside execute job as we are getting data, but imo its ok to do that in a follow up PR, while keeping this PR as small as possible and just abstracting our storage without doing any other changes.

Reply:
I have addressed that in the latest commit. The task runner functions no longer return content but write that to the stream.